### PR TITLE
pb-3798: The progress bar to show total count correctly

### PR DESCRIPTION
**Pull request by lalat-das**
- In case of multi namespace scenario the show status picking the total resource count before the filteration when we restore only selective namesapces
- Hence updated the total resource count at the appropriate place so that multi namespace backup's restore with a iselective namespaces to show correct count

**What type of PR is this?** Bug
> Uncomment only one and also add the corresponding label in the PR:
bug
>feature
>improvement
>cleanup
>api-change
>design
>documentation
>failing-test
>unit-test
>integration-test

**What this PR does / why we need it**: 

**Does this PR change a user-facing CRD or CLI?**: - In case of multi namespace scenario the show status picking the total resource count before the filteration when we restore only selective namesapces
- Hence updated the total resource count at the appropriate place so that multi namespace backup's restore with a iselective namespaces to show correct count

**Is a release note needed?**: no

```release-note
Issue:
User Impact:
Resolution

```

**Does this change need to be cherry-picked to a release branch?**: 23.3.1

Unit tested details are shown in pb-3798 bug report

### CODE REVIEW

**Review by siva-portworx — APPROVED** [review]

**Review by px-kesavan — APPROVED** [review]
LGTM

**Comment by siva-portworx** [comment]
Force pushing PR as QA is blocked on this. Will track after merging.

**Pull request merged**
